tests/chainer_tests/functions_tests/array_tests/test_cast.py

```diff
@@ -57,4 +57,21 @@ def test_backward_cpu(self):
         self.check_backward(self.x, self.g)
 
 
+class TestNoCast(unittest.TestCase):
+
+    def setUp(self):
+        self.dtype = numpy.float32
+        self.x = numpy.empty(1, self.dtype)
+
+    def check_forward_no_cast(self, x_data):
+        y = functions.cast(x_data, self.dtype)
+        self.assertIs(y, x_data)
```

Comment:
I mean consistency. If the function returns both `ndarray` and `Variable`, a user needs to check the type of the return value. Even when only parts of functions behaves like that, a user needs to check its specification precisely.
I feel it is better that all functions in `chainer.functions` return `Variable`. How do you think?